library/extractor/src/main/java/com/google/android/exoplayer2/metadata/id3/Id3Decoder.java

```diff
@@ -474,15 +475,38 @@ private static TextInformationFrame decodeTextInformationFrame(
     }
 
     int encoding = id3Data.readUnsignedByte();
-    String charset = getCharsetName(encoding);
 
     byte[] data = new byte[frameSize - 1];
     id3Data.readBytes(data, 0, frameSize - 1);
 
-    int valueEndIndex = indexOfEos(data, 0, encoding);
-    String value = new String(data, 0, valueEndIndex, charset);
+    List<String> values = decodeTextInformationFrameValues(data, encoding, 0);
+    return new TextInformationFrame(id, null, values);
+  }
+
+  @NonNull
+  private static List<String> decodeTextInformationFrameValues(
+      byte[] data, final int encoding, final int index) throws UnsupportedEncodingException {
+    ArrayList<String> values = new ArrayList<>();
+
+    if (index >= data.length) {
+      return Collections.emptyList();
+    }
+
+    // In ID3v2.4, text information frames can contain multiple values delimited by a null
+    // terminator. Thus, we after each "end of stream" marker we actually need to keep looking
+    // for more data, at least until the index is equal to the data length.
```

Comment:
I think you can remove this comment too. IMO it only really makes sense as part of the 'diff' from the previous (current) behaviour, but that diff won't be visible to a reader in future (e.g. in 1 year). We should just assume that multiple values are 'natural' so no need to explicitly comment it.